src/parser/core/modules/CastTime.ts

```diff
@@ -31,52 +32,128 @@ export default class CastTime extends Analyser {
+	 */
+	public setPercentageAdjustmentFflogs(actions: number[] | 'all', reduction: number, start: number = this.parser.currentTimestamp, end: number | null = null): number {
+		return this.setPercentageAdjustment(actions, reduction, this.parser.fflogsToEpoch(start), end ? this.parser.fflogsToEpoch(end) : null)
+	}
+	/**
+	 * Sets a cast time adjustment for a percentage change per cast (See: Swiftcast, RDM's Doublecast trait, Ley Lines, etc.)
+	 * @param actions The actions this adjustment applies to. Either an array of IDs, or the string 'all'
+	 * @param adjustment The percentage multiplier to adjust the cast time to (ie 0 for instant cast, 0.85 for Ley Lines, 1.25 for a 25% slow)
+	 * @param start The beginning of the adjustment time range. Defaults to the current epoch timestamp
+	 * @param end The end of the adjustment time range. May be left null if the end of the range is not yet known
+	 * @returns The index number within the cast time adjustments collection, can be used to reset/end this adjustment later
+	 */
+	public setPercentageAdjustment(actions: number[] | 'all', adjustment: number, start: number = this.parser.currentEpochTimestamp, end: number | null = null): number {
+		return this.set(actions, 'percentage', Math.max(adjustment, 0), start, end)
+	}
+	private set(actions: number[] | 'all', type: 'time' | 'percentage', adjustment: number, start: number = this.parser.currentEpochTimestamp, end: number | null = null): number {
 		const newLength = this.castTimes.push({
 			actions,
-			castTime,
+			type,
+			adjustment,
 			start,
 			end,
 		})
 
 		return newLength - 1
 	}
 
-	/** @deprecated */
+	/**
+	 * reset Module-compatibility function
+	 * @deprecated
+	 */
 	public resetFflogs(id: number | null, timestamp: number = this.parser.currentTimestamp): void {
 		this.reset(id, this.parser.fflogsToEpoch(timestamp))
 	}
+	/**
+	 * Sets the 'end' property of the specified cast time reduction range
+	 * @param id The index within the cast times adjustment collection (provided by the set functions)
+	 * @param timestamp The timestamp at which the reduction range ended. Defaults to the current epoch timestamp
+	 */
 	public reset(id: number | null, timestamp = this.parser.currentEpochTimestamp): void {
 		if (id == null) { return }
 		const ct = this.castTimes[id]
 		if (!ct) { return }
 		ct.end = timestamp
 	}
 
-	/** @deprecated */
+	/**
+	 * forEvent Module-compatibility function
+	 * @deprecated
+	 */
 	public forFflogsEvent(event: CastEvent): number | undefined {
 		return this.forAction(event.ability.guid, this.parser.fflogsToEpoch(event.timestamp))
 	}
+	/**
+	 * Returns the effective cast time for the specified event
+	 * @param event The event in question
+	 * @returns The actual cast time, either as the default, or the modified time if any modifiers were in effect. Returns undefined if the action cannot be determined, or has no default cast time
+	 */
 	public forEvent(event: Events['action']): number | undefined {
```

Comment:
random thought in passing; while i don't really see the point in this method any more, probably worth accepting `prepare` as well if we keep it.

Reply:
Weaving's using the (currently fflogs-version) forEvent function to check if the leading GCD event for a weave was instant. Probably not a bad idea to accept prepare as well though. Done.